File: src/test/java/br/com/leonardoferreira/jirareport/factory/UpdateBoardRequestFactory.java
package br.com.leonardoferreira.jirareport.factory;

import br.com.leonardoferreira.jbacon.JBacon;
import br.com.leonardoferreira.jirareport.domain.ImpedimentType;
import br.com.leonardoferreira.jirareport.domain.request.UpdateBoardRequest;
import br.com.leonardoferreira.jirareport.domain.vo.DynamicFieldConfig;
import com.github.javafaker.Faker;
import java.util.Arrays;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Component;

@Component
public class UpdateBoardRequestFactory extends JBacon<UpdateBoardRequest> {

    @Autowired
    private Faker faker;

    @Override
    protected UpdateBoardRequest getDefault() {
        UpdateBoardRequest updateBoardRequest = new UpdateBoardRequest();

        updateBoardRequest.setStartColumn(faker.lorem().word());
        updateBoardRequest.setEndColumn(faker.lorem().word());
        updateBoardRequest.setFluxColumn(faker.lorem().words());
        updateBoardRequest.setName(faker.lorem().word());
        updateBoardRequest.setIgnoreIssueType(faker.lorem().words());
        updateBoardRequest.setEpicCF(faker.expression("custom_field_#{number.number_between '1000','9999'}"));
        updateBoardRequest.setEstimateCF(faker.expression("custom_field_#{number.number_between '1000','9999'}"));
        updateBoardRequest.setSystemCF(faker.expression("custom_field_#{number.number_between '1000','9999'}"));
        updateBoardRequest.setProjectCF(faker.expression("custom_field_#{number.number_between '1000','9999'}"));
        updateBoardRequest.setCalcDueDate(faker.random().nextBoolean());
        updateBoardRequest.setIgnoreWeekend(faker.random().nextBoolean());
        updateBoardRequest.setImpedimentType(faker.options().option(ImpedimentType.class));
        updateBoardRequest.setImpedimentColumns(faker.lorem().words());
        updateBoardRequest.setDynamicFields(Arrays.asList(
                new DynamicFieldConfig("dnf_1", faker.expression("custom_field_#{number.number_between '1000','9999'}")),
                new DynamicFieldConfig("dnf_2", faker.expression("custom_field_#{number.number_between '1000','9999'}")),
                new DynamicFieldConfig("dnf_3", faker.expression("custom_field_#{number.number_between '1000','9999'}"))
        ));
        updateBoardRequest.setTouchingColumns(faker.lorem().words());
        updateBoardRequest.setWaitingColumns(faker.lorem().words());

        return updateBoardRequest;
    }

    @Override
    protected UpdateBoardRequest getEmpty() {
        return new UpdateBoardRequest();
    }

    @Override
    protected void persist(final UpdateBoardRequest updateBoardRequest) {
        throw new UnsupportedOperationException();
    }

}

File: src/test/java/br/com/jiratorio/factory/domain/request/UpdateBoardRequestFactory.kt
package br.com.jiratorio.factory.domain.request

import br.com.jiratorio.domain.duedate.DueDateType
import br.com.jiratorio.domain.dynamicfield.DynamicFieldConfig
import br.com.jiratorio.domain.impediment.ImpedimentType
import br.com.jiratorio.domain.request.UpdateBoardRequest
import br.com.jiratorio.extension.faker.jira
import br.com.leonardoferreira.jbacon.JBacon
import com.github.javafaker.Faker
import org.springframework.stereotype.Component
import java.util.Arrays.asList

@Component
class UpdateBoardRequestFactory(
    private val faker: Faker
) : JBacon<UpdateBoardRequest>() {

    override fun getDefault() =
        UpdateBoardRequest(name = faker.lorem().word()).apply {
            startColumn = faker.lorem().word()
            endColumn = faker.lorem().word()
            fluxColumn = faker.lorem().words()
            ignoreIssueType = faker.lorem().words()
            epicCF = faker.jira().customField()
            estimateCF = faker.jira().customField()
            systemCF = faker.jira().customField()
            projectCF = faker.jira().customField()
            ignoreWeekend = faker.random().nextBoolean()
            impedimentType = faker.options().option(ImpedimentType::class.java)
            impedimentColumns = faker.lorem().words()
            dynamicFields = asList(
                DynamicFieldConfig("dnf_1", faker.jira().customField()),
                DynamicFieldConfig("dnf_2", faker.jira().customField()),
                DynamicFieldConfig("dnf_3", faker.jira().customField())
            )
            touchingColumns = faker.lorem().words()
            waitingColumns = faker.lorem().words()
            dueDateCF = faker.jira().customField()
            dueDateType = faker.options().option(DueDateType::class.java)
        }

    override fun getEmpty() =
        UpdateBoardRequest(name = faker.lorem().word())

    override fun persist(updateBoardRequest: UpdateBoardRequest) =
        throw UnsupportedOperationException()

}
